### DISCUSSION TITLE
v2.7 ApiResource

### DISCUSSION THREAD

**Discussion by asispts** [api-platform/core#4852] [Q&A] [answered]
Hello

I'm facing a bug in v2.6 and trying to upgrade to v2.7 to see if it's still exist. However, after replacing `ApiPlatform\Core\Annotation\ApiResource` with `ApiPlatform\Metadata\ApiResource`, I got nothing.

![nothing](https://user-images.githubusercontent.com/79239132/181702739-c512d84f-0a28-4fe4-a1bb-ddeb5a632031.png)

Use this code or another one will give the same result.
```php
<?php

use ApiPlatform\Metadata\ApiResource;
use Doctrine\ORM\Mapping as ORM;

#[ApiResource]
#[ORM\Entity]
class Entity
{
```
```php
<?php

use ApiPlatform\Metadata\ApiResource;
use ApiPlatform\Metadata\Get;
use ApiPlatform\Metadata\GetCollection;
use Doctrine\ORM\Mapping as ORM;

#[ApiResource(operations:[
    new Get(),
    new GetCollection()
])]
#[ORM\Entity]
class Entity
{
```

Did I miss something?

```
PHP: 8.1
symfony: 6.1
api-platform: 2.7.0-rc.2
```

**Comment by tacman** [comment]
Try clearing everything in the cache:

bin/console cache:pool:clear cache.serializer cache.annotations
cache.property_info \
  api_platform.cache.route_name_resolver \
  api_platform.cache.metadata.resource\
  api_platform.cache.metadata.property \
  api_platform.cache.metadata.resource_collection

bin/console cache:clear

**Reply by asispts** [reply]
I did, even removed the `var` directory, nothing changed.

I will try to replicate the issue on `php 7.4` and `symfony 5.4`.

**Comment by jamesisaac** [comment]
Make sure you have added `metadata_backward_compatibility_layer: false` to `api_platform.yaml`, as noted in https://github.com/api-platform/docs/blob/main/core/upgrade-guide.md#i-want-to-try-the-new-metadata-system

**Comment by asispts** [comment, marked as answer]
Thank you @jamesisaac 

Adding `metadata_backward_compatibility_layer: false` and make sure PHP version is `8.1` fix the issue.
